### PR TITLE
Application Gateway Key Vault Certificates

### PR DESCRIPTION
**Pull request by Gordonby**
# PR Checklist

New template that shows leveraging the kv-cert module from the public bicep registry. The generated certificates are used in an Application Gateway which is created with a sample app to show SSL termination.

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### CODE REVIEW

**Comment by Gordonby** [comment]
Yay it passed 😄
image

**Comment by Gordonby** [comment]
Because of the propagation error messages, I've had to introduce an explicit delay (wait.bicep) in order for the deployments to consistently suceed.

**Comment by sahmed13057** [comment]
Can anyone help me with devops server silent installation?
thanks

**Comment by Gordonby** [comment]
> Can anyone help me with devops server silent installation? thanks

I'd ask this question in the [Technical Community Forum](https://techcommunity.microsoft.com/t5/azure-devops/bd-p/AzureDevOpsForum) Not this pull request 😉
Or if you'd like to watch a comical animated gif of me click-ops'ing the install then look [here](https://animated.azurewebsites.net/azuredevopsserver.htm)

File: quickstarts/microsoft.resources/deployment-script-azcli-agw-certificates/appgw.bicep [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+  sku: {
+    name: 'Standard'
+  }
+  properties: {
+    publicIPAllocationMethod: 'Static'
+  }
+}
+output agwPip string = appgwpip.properties.ipAddress
+
+var keyVaultSecretsUserRole = resourceId('Microsoft.Authorization/roleDefinitions', '4633458b-17de-408a-b874-0445c86b69e6')
```
Comment (bmoore-msft) [inline quickstarts/microsoft.resources/deployment-script-azcli-agw-certificates/appgw.bicep:183]: move all vars to the same section of the file

File: quickstarts/microsoft.resources/deployment-script-azcli-agw-certificates/main.bicep [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      family: 'A'
+      name: 'standard'
+    }
+    tenantId: subscription().tenantId
+    enableSoftDelete: false
+    enableRbacAuthorization: true
+    accessPolicies: []
+  }
+}
+output akvName string = akv.name
```
Comment (bmoore-msft) [inline quickstarts/microsoft.resources/deployment-script-azcli-agw-certificates/main.bicep:61]: organize the `outputs` to the end of the file and `vars` between `params` and `resources`

File: quickstarts/microsoft.resources/deployment-script-azcli-agw-certificates/appgw.bicep [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+  scope: akv
+  name: guid(resourceId('Microsoft.Network/applicationGateways',agwName), agwId.id, akv.id, keyVaultSecretsUserRole)
+  properties: {
+    roleDefinitionId: keyVaultSecretsUserRole
+    principalType: 'ServicePrincipal'
+    principalId: agwId.properties.principalId
+  }
+}
+
+module rbacPropagationDelay 'wait.bicep' = {
```
Comment (bmoore-msft) [inline quickstarts/microsoft.resources/deployment-script-azcli-agw-certificates/appgw.bicep:194]: why is this needed? since the AGW assignment has already taken place?  What step is left to "fail"?

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: quickstarts/microsoft.resources/deployment-script-azcli-agw-certificates/aciApp.bicep [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  location: location
+  properties: {
+    containers:  [
+      {
+        name: 't1'
+        properties: {
+          image: appImage
+          resources: {
+            requests: {
+              cpu: json('1')
```
[the remark this answers is not in the document]
Comment (Gordonby) [reply quickstarts/microsoft.resources/deployment-script-azcli-agw-certificates/aciApp.bicep:21]: bicep complains that 1.5 is not an integer. the accepted workaround seems to be to wrap in a json().
i've changed the bicep code so this isn't a problem anymore

**Comment by Gordonby** [comment]
Linking this PR to my module proposal in the bicep registry which will abstract the wait.bicep pattern

https://github.com/Azure/bicep-registry-modules/issues/114
